### PR TITLE
[Unity][TIR] Clear struct info when specializing PrimFunc

### PR DESCRIPTION
**Pull request by Lunderberg**
In rare cases, a `PrimFunc` may be annotated with `StructInfo`, to indicate that it is an impure function with specific shapes for the parameters.  If struct info is present, it is invalidated when specializing a `PrimFunc`, and should be cleared.

### CODE REVIEW

File: tests/python/tir-base/test_tir_specialize.py [hunk 1]
```diff
@@ -324,5 +326,41 @@ def expected(A_data: T.handle("float32")):
     tvm.ir.assert_structural_equal(expected, after)
 
 
+def test_specialization_removes_struct_info():
+    """Reset struct info in specialization
+
+    While a PrimFunc usually doesn't have a `relax.StructInfo`, the
+    field can be populated in some edge cases.  If that PrimFunc is
```
Comment (slyubomirsky) [inline tests/python/tir-base/test_tir_specialize.py:333]: Do you know of times when we're doing it? It should be something we do consistently or not at all.

**Review by slyubomirsky — COMMENTED** [review]
The change seems straightforward but I would be more curious to hear about why it's necessary (i.e., what problems were being caused) and when the StructInfo is being set for PrimFuncs.

File: tests/python/tir-base/test_tir_specialize.py (hunk 1 above)
Comment (Lunderberg) [reply tests/python/tir-base/test_tir_specialize.py:333]: There's a couple of MLC ([example](https://github.com/mlc-ai/mlc-llm/blob/fb4c52c749c4b32594f1e9382113fecd11847893/mlc_llm/transform/fuse_split_rotary_embedding.py#L100)) where the a dynamic PrimFunc is annotated with the `relax.FuncStructInfo`.  This allows the `R.call_tir` shape inference to work correctly, even after the arguments are mutated.

IIRC, the primary reasons not to add `FuncStructInfo` to PrimFuncs was that primfuncs didn't follow the relax requirements for functional purity.  Since we now have the purity expressed within `FuncStructInfo`, I think that means we could add `FuncStructInfo` to every PrimFunc on construction, without making any incorrect statements about the functions.

Comment (slyubomirsky) [reply tests/python/tir-base/test_tir_specialize.py:333]: Always specifying it sounds like a better approach, as long as it would not interfere with anything in TIR.
